File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int res = 0;

        for(auto i:nums){
            if(numSet.find(i-1)==numSet.end()){
                int length = 1;
                while(numSet.find(i+length)!=numSet.end()){
                    length++;
                }
                res = max(res, length);
            }
        }

        return res;

    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        nums.sort()

        if len(nums) == 0:
            return 0
        
        count = 1
        maxCount = 1
        for i in range (1, len(nums)):
            
            if (nums[i] == nums[i-1]+1):
                count += 1
            
            elif (nums[i] == nums[i-1]):
                continue
            
            else:
                count = 1

            maxCount = max(maxCount, count)
        

        return maxCount